src/index.js

```diff
@@ -0,0 +1,196 @@
+import './draggableList.less'
+
+const React = require('react')
+
+const DEFAULT_DRAG_THRESHOLD = 10
+const RESTRICT_DRAGGING = true
+const MATCH_MOUSE_DOWN_BUTTON = 0
+
+module.exports = React.createClass({
+  displayName: 'DraggableList',
+  propTypes: {
+    onOrderChanged: React.PropTypes.func.isRequired,
+    items: React.PropTypes.array.isRequired,
+    dragStartThreshold: React.PropTypes.number,
+    restrictDragging: React.PropTypes.bool,
+    className: React.PropTypes.string,
+    previewClassName: React.PropTypes.string,
+    matchMouseDownButton: React.PropTypes.number
+  },
+
+  /* Parameters access */
+  getDragThreshold () {
+    return this.props.dragStartThreshold || DEFAULT_DRAG_THRESHOLD
+  },
+  getRestrictDragging () {
+    return this.props.restrictDragging || RESTRICT_DRAGGING
+  },
+  getMatchMouseDownButton () {
+    return this.props.matchMouseDownButton || MATCH_MOUSE_DOWN_BUTTON
+  },
+
+  /* Lifecycle */
+  componentDidMount () {
+    document.addEventListener('mouseup', this.onMouseUp)
```

Comment:
Why do you listen to document as opposed to the root element (for instance)?

Reply:
I need those events to trigger even if the mouse is no longer over the component.